spec/tapioca/compilers/dsl/url_helpers_spec.rb

```diff
@@ -0,0 +1,293 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe("Tapioca::Compilers::Dsl::UrlHelpers") do
+  before(:each) do
+    require "tapioca/compilers/dsl/url_helpers"
+  end
+
+  subject do
+    Tapioca::Compilers::Dsl::UrlHelpers.new
+  end
+
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    content = <<~RUBY
+      class Application < Rails::Application
+      end
+    RUBY
+
+    it("gathers constants that include path_helpers_module") do
+      content += <<~RUBY
+        class MyClass
+          include Rails.application.routes.named_routes.path_helpers_module
```

Comment:
Let's always work with `Rails.application.routes.url_helpers`, since that is the actual thing we are trying to address.

You can use `include/extend Rails.application.routes.url_helpers(true/false)` to test various combinations.